client/app/scripts/components/node-details/node-details-table.js

```diff
@@ -1,33 +1,159 @@
+import _ from 'lodash';
 import React from 'react';
 
-import NodeDetailsTableRowValue from './node-details-table-row-value';
-import NodeDetailsTableRowNumber from './node-details-table-row-number';
-import NodeDetailsTableRowSparkline from './node-details-table-row-sparkline';
+import NodeDetailsTableNodeLink from './node-details-table-node-link';
+import { formatMetric } from '../../utils/string-utils';
 
 export default class NodeDetailsTable extends React.Component {
+
+  constructor(props, context) {
+    super(props, context);
+    this.DEFAULT_LIMIT = 5;
+    this.state = {
+      limit: this.DEFAULT_LIMIT,
+      sortedDesc: true,
+      sortBy: null
+    };
+    this.handleLimitClick = this.handleLimitClick.bind(this);
+    this.getValueForSortBy = this.getValueForSortBy.bind(this);
+  }
+
+  handleHeaderClick(ev, headerId) {
+    ev.preventDefault();
+    const sortedDesc = headerId === this.state.sortBy ? !this.state.sortedDesc : this.state.sortedDesc;
+    const sortBy = headerId;
+    this.setState({sortBy, sortedDesc});
+  }
+
+  handleLimitClick(ev) {
+    ev.preventDefault();
+    const limit = this.state.limit ? 0 : this.DEFAULT_LIMIT;
+    this.setState({limit});
+  }
+
+  getDefaultSortBy() {
+    // first metric
+    const nodes = this.props.nodes;
+    if (nodes && nodes.length > 0) {
+      if (nodes[0].metrics && nodes[0].metrics.length > 0) {
+        return nodes[0].metrics[0].id;
+      }
+    }
+    return null;
+  }
+
+  getMetaDataSorters() {
+    // returns an array of sorters that will take a node
+    if (this.props.nodes && this.props.nodes.length > 0 && this.props.nodes[0].metadata) {
+      return this.props.nodes[0].metadata.map((field, index) => {
+        return node => node.metadata[index] ? node.metadata[index].value : null;
```

Comment:
more fun w/ _.get!
`return _.get(this.props.nodes, [0, 'metadata'], []).map(...);`